### DISCUSSION TITLE
Astro: Is there a way to autofill `defineCollection` based on the keystatic config?

### DISCUSSION THREAD

**Discussion by tordans** [Thinkmill/keystatic#1143] [Q&A]
It looks like ATM I have to define my data structure twice when using Astro: once as keystatic `config` and once in `/content/config.ts` with `defineCollection` ([following this doc](https://docs.astro.build/en/guides/content-collections/)).

Is there a way to populate the Astro `defineCollection` based on the keystatic `config` so both hat I only have to manage one?

**Comment by smokeyfro** [comment]
You could try using this -> https://github.com/gxjansen/Astro-CC2KS

**Reply by tordans** [reply]
Thanks for sharing. It works the other way around than what I was looking for, but I will test it out some time.

In my mind, the astro config is a subset of the keystatic config in the way that keystatic knows more about the raw data, which is why I was thinking about it the other way around, create the astro config from the keystatic config.
